tests/io/test_data_catalog.py

```diff
@@ -953,13 +949,13 @@ def test_sorting_order_with_other_dataset_through_extra_pattern(
         )
         sorted_keys_expected = [
             "{country}_companies",
-            "{another}#csv",
             "{namespace}_{dataset}",
             "{dataset}s",
+            "{another}#csv",
```

Comment:
How come the sorting order changed?

Reply:
Currently, we have `dataset_patterns`, `default_pattern` and `runtime_patterns`, which are sorted and stored separately. They resolved in the following order: dataset_patterns`, `default_pattern`, `runtime_patterns`.

Previously, we merged all of them into the `dataset_patterns` and sorted it so patterns from `runtime_patterns` could be resolved earlier than those presented in the `dataset_patterns` from the beginning. Now, it's impossible. The current test checks this case. Since `runtime_patterns` are sorted separately and resolved at the last place, the order has changed.

This refactored approach was discussed with @ankatiyar, and we concluded that it is the right way to resolve them.

Reply:
That makes sense. Thanks for explaining!